sanic/server/websockets/connection.py

```diff
@@ -0,0 +1,82 @@
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Dict,
+    List,
+    MutableMapping,
+    Optional,
+    Union,
+)
+
+
+ASIMessage = MutableMapping[str, Any]
+
+
+class WebSocketConnection:
+    """
+    This is for ASGI Connections.
+    It provides an interface similar to WebsocketProtocol, but
+    sends/receives over an ASGI connection.
+    """
+
+    # TODO
+    # - Implement ping/pong
```

Comment:
What's the story here?

Reply:
This isn't my code, it is the ASGI-websocket wrapper/adapter class. It was moved from [here](https://github.com/sanic-org/sanic/blob/404c5f9f9eb54f06d1f30250bc2cc840c18a0eab/sanic/websocket.py#L146). I don't know if it makes sense for Sanic to do the ping/pong handling for an ASGI websocket, or perhaps the ASGI server handles that itself.